python/ray/actor.py

```diff
@@ -273,8 +261,21 @@ def __init__(self, modified_class, class_id, checkpoint_interval, num_cpus,
         self._actor_method_cpus = actor_method_cpus
         self._exported = False
 
+        # Assign an __init__ function will avoid many checks later on.
+        def __init__(self):
+            pass
         self._actor_methods = inspect.getmembers(
             self._modified_class, ray.utils.is_function_or_method)
+        self._actor_method_names = [
+            method_name for method_name, _ in self._actor_methods
+        ]
+        if "__init__" not in self._actor_method_names:
```

Comment:
yep, that'd be cleaner. we can also define the local `__init__` function in the if section, because it won't be needed otherwise.

Reply:
I have test `hasattr(self._modified_class, "__init__"):`. There will some test test case failure. If a class is derived from `object`, it will have a `__init__` function from object. In this case, the `hasattr(self._modified_class, "__init__")` will be true, but `_method_signatures` will not contain "__init__". 
```python
@ray.remote
class Foo(object):
def method(self):
    pass
```